x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/json_diff.test.tsx

```diff
@@ -0,0 +1,290 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { render, within, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import type { Matcher } from '@testing-library/react';
+import { escapeRegExp } from 'lodash';
+
+import { RuleDiffTab } from '../rule_diff_tab';
+import { savedRuleMock } from '../../../logic/mock';
+import type { RuleResponse } from '../../../../../../common/api/detection_engine/model/rule_schema/rule_schemas.gen';
+
+/**
+ * Creates a matcher function designed to determine if an element contains specific text content in specific order.
+ * It returns true if the element itself contains the strings (matching all strings in the specified order) without
+ * any of its children containing all of these strings. This is useful for finding elements with text content that
+ * is split across multiple child elements.
+ *
+ * @param {string[]} strings - An array of strings to be matched within the content.
+ * @returns {Function} A Matcher function that returns a boolean indicating whether the element matches the
+ * specified text conditions: the element contains the text specified by `strings` and none of its children
+ * contain this text.
+ *
+ * The Matcher function itself has the following signature:
+ * @param {string} content - The content string to be checked. Not directly used in the current implementation
+ * but is required for the matcher's signature.
+ * @param {Element|null} element - The DOM element to check for the text content. If null, the matcher will
+ * return false.
+ *
+ * @returns {boolean} `true` if the element contains the specified text and none of its children do,
+ * otherwise `false`.
+ * Utilizes Lodash's `escapeRegExp` to handle special characters in the `strings` array.
+ */
+const matchInOrder =
```

Comment:
Have you considered using `Diff` component's `textContent` instead?

I'm afraid this helper will make the test logic harder but it still doesn't protect again sensitivity to changes. Looking at the Diff's `textContent` below it's clear we should be able to implement many testing scenarios.

`{{  "author": [  "author": [    "Alice",    "Alice",-    "Bob",    "Charlie"    "Charlie"  ],  ],  "description": "some desc",  "description": "some desc",Expand 15 unchanged lines  ],  ],  "interval": "5m",  "interval": "5m",  "language": "kuery",  "language": "kuery",+  "license": "GPLv3",  "max_signals": 100,  "max_signals": 100,  "name": "Test rule",  "name": "Test rule",  "note": "",  "note": "",Expand 19 unchanged lines  ],  ],  "to": "now",  "to": "now",  "type": "query",  "type": "query",-  "version": 1+  "version": 2}}`

For example normalizing it by removing extra spaces will make it simple to implement a test which verifies there is `-  "version": 1+  "version": 2`. Of course it can break as well 

We also shouldn't retest `react-diff-view`. It's enough to make sure the content is conceptually the same but how may components/tags and etc used isn't so important and will make the tests fragile.

Reply:
Refactored this part in a follow-up commit